lib/io/byte-order.toit

```diff
@@ -26,6 +26,41 @@ A byte order class with support for common read and write operations on byte
   arrays.
 */
 abstract class ByteOrder:
+  /**
+  Swaps the byte-order of all 16-bit integers in the $byte-array.
+  If the integers were in little-endian order they then are in big-endian byte order
+  If the integers were in big-endian order they then are in little-endian byte order.
+  */
+  swap-16 byte-array/ByteArray -> none:
+    if byte-array.size <= 8:
+      for i := 0; i < byte-array.size; i += 2:
+        value := LITTLE-ENDIAN.uint16 byte-array i
+        BIG-ENDIAN.put-uint16 byte-array i value
+      return
+    tmp := ByteArray
+      max byte-array.size 512
+    List.chunk-up 0 byte-array.size 512: | from to size |
+      blit byte-array[from + 1..to] tmp[0..size] size/2 --source-pixel-stride=2 --destination-pixel-stride=2
+      blit byte-array[from    ..to] tmp[1..size] size/2 --source-pixel-stride=2 --destination-pixel-stride=2
+      byte-array.replace from tmp 0 size
+
+  /**
+  Swaps the byte-order of all 32-bit integers in the $byte-array.
+  If the integers were in little-endian order they then are in big-endian byte order
+  If the integers were in big-endian order they then are in little-endian byte order.
+  */
+  swap-32 byte-array/ByteArray -> none:
+    tmp := ByteArray
+      max byte-array.size 512
+    List.chunk-up 0 byte-array.size 512: | from to size |
+      slice := byte-array[from..to]
+      buffer := tmp[..size]
+      blit slice[3..] buffer      size/4 --source-pixel-stride=4 --destination-pixel-stride=4
```

Comment:
Cache `size/4` in a local.

Reply:
Will do this in a follow-up PR. Here I just move the functions.